noisy.py

```diff
@@ -239,6 +239,7 @@ def crawl(self):
 
             except requests.exceptions.RequestException:
                 logging.warn("Error connecting to root url: {}".format(url))
+                time.sleep(random.randrange(self._config["min_sleep"], self._config["max_sleep"]))
```

Comment:
Maybe add some docs explaining what issue it's trying to solve?
Also, I think it would be better to handle this by catching the `requests.exceptions.ConnectionError` exception.

> In the event of a network problem (e.g. DNS failure, refused connection, etc), Requests will raise a ConnectionError exception.

Reply:
Exceptions handling starts to get lengthy indeed!

The problem I faced was that the internet connection dropped for a few minutes. Noisy could not connect to the URLs and started looping at a very fast pace (~100 per second) on the root urls, resulting in logs clogged with 36K + warnings. I thought relevant to add the same sleep mechanism than for successful requests. Detecting ConnectionError and addressing it specifically to avoid the log runaway does of course makes sense too.